pkg/lib/usage/globaldb_store.go

```diff
@@ -214,6 +215,19 @@ func (s *GlobalDBStore) FetchUsageRecordsInRange(
 	fromStartTime time.Time,
 	toEndTime time.Time,
 ) ([]*UsageRecord, error) {
+	// end_time is not indexed, so we want to query by start_time
+	// construct a query which is effectively `start_time >= ? AND end_time <= ?`
+	var endStartTime time.Time
+	switch period {
+	case periodical.Hourly:
+		endStartTime = toEndTime.Add(-time.Hour)
+	case periodical.Daily:
+		endStartTime = toEndTime.Add(-time.Hour * 24)
+	case periodical.Weekly:
+		endStartTime = toEndTime.Add(-time.Hour * 24 * 7)
+	case periodical.Monthly:
+		endStartTime = timeutil.FirstDayOfTheMonth(toEndTime).Add(-time.Hour * 24)
+	}
```

Comment:
I still prefer `start_time >= ? AND start_time < ?`. This is more aligned with the definition of the column. `start_time` is inclusive and `end_time` is exclusive. To convert the inclusive `toEndTime`, we should add 1 `period` to it.

Reply:
I  guess you mean `toEndTime` of the function should be exclusive. However this makes it difficult to use, you need to add a little moment to `toEndTime` (e.g. 1 seconds) to query the rows you want.

Reply:
No, `toEndTime` can be inclusive. I mean how to convert the inclusive Go value `toEndTime` to a value suitable for use in SQL.

Reply:
Actually I found that the new implementation is not very intuitive which makes me prefer the previous implementation. I also thinks that the performance gain for this change is not very important.

Reply:
> No, toEndTime can be inclusive. I mean how to convert the inclusive Go value toEndTime to a value suitable for use in SQL.

If you mean calculate `endStartTime` by minus 1 period from `toEndTime`, it is already the current implementation. However, we cannot do `start_time >= ? AND start_time < ?`. We need `start_time <= ?` so that all needed rows are returned, because `end_time` minus 1 period is exactly the `start_time`.

Reply:
Removed conversion of toEndTime in latest commit and used it to filter start_time exclusively.